sdk/src/azure/iot/az_iot_hub_client.c

```diff
@@ -74,6 +79,11 @@ AZ_NODISCARD az_result az_iot_hub_client_get_user_name(
   {
     required_length += az_span_size(*user_agent) + az_span_size(hub_client_param_separator_span);
   }
+  if (az_span_size(*model_id) > 0)
+  {
+    required_length += az_span_size(hub_client_param_separator_span)
+        + az_span_size(hub_client_param_equals_span);
```

Comment:
Missing + az_span_size(*model_id) ?

Reply:
I think that's up by line 76 but I'm not seeing adding the size for `hub_digital_twin_model_id` anywhere during initial calc.  If(!!) this is indeed a bug would be good to understand how it got past tests.

Reply:
added comment for why it's missing

Reply:
added comment for why it's missing